File: mobile/lib/enums/tipo_granja_constants.dart
// lib/enums/tipo_granja_constants.dart

/// Enum representando o id referente os tipos de granja.
enum TipoGranjaId {
  gestacao,
  geral,
  creche,
  inseminacao
}

/// Mapeamento de TipoGranjaId para valores inteiros.
const Map<TipoGranjaId, int> tipoGranjaIdToInt = {
  TipoGranjaId.gestacao: 1,
  TipoGranjaId.geral: 2,
  TipoGranjaId.creche: 3,
  TipoGranjaId.inseminacao: 4,
};

File: API/src/constants/tipoGranjaConstants.ts
// src/constants/tipoGranjaConstants.ts

export enum TipoGranjaId {
    GESTACAO = 1,
    GERAL = 2,
    CRECHE = 3,
    INSEMINACAO = 4,
}

export const TipoGranjaIdDescriptions: Record<TipoGranjaId, string> = {
    [TipoGranjaId.GESTACAO]: 'Granja para gestação de porcas',
    [TipoGranjaId.GERAL]: 'Granja para uso geral',
    [TipoGranjaId.CRECHE]: 'Granja para crechário de leitões',
    [TipoGranjaId.INSEMINACAO]: 'Granja para inseminação de porcas',
};
